### PR TITLE
Spike/mle abstract tailwind imports

### PR DESCRIPTION
**Pull request by michealengland**
Experimental Solution for: #614 

### DESCRIPTION

Even though we're likely going a different direction, this spike branch does the following:

1. Abstract Tailwind into a separate webpack entry.
2. Creates `style.scss` and `editor.scss` for importing styles
3. Update enqueues similar to #600 since these updates are very tightly coupled with this update
4. Add `@tailwind base;` to `style.scss` and `editor.scss` so that we can control the behavior behind this feature and address some concerns around selector name scoping.

See `spike/mle-abstract-tailwind-imports`

### Things not setup in this branch:
- Selector namespace updates are not part of this PR. This was part of the discussion but is a different issue. Importing `@tailwind base;` will make this issue easier to resolve and iterate on.
- CSS Mapping in the editor `DevTools failed to load SourceMap: Could not load content for http://wdsunderscores.test/wp-admin/index.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE` 

**Results**
_Times will vary depending on your machine._

- Rebuild times appear to range from `3500ms-4000ms` on my local using my MBP. Which is not much more than before these changes. Times will vary depending on what was change for the rebuild. On my Mac M1, rebuild times are around `2000ms-2300ms`.
- Regular build times appear to be around the same time.

### SCREENSHOTS
Examples below are taken from my Mac Mini M1. Results WILL vary according to you machine.

Editor styles being loaded with `@apply bg-purple-500;` on `.editor-post-title `.
Screen Shot 2021-03-26 at 9 55 01 AM

Rebuild example when adding style `@apply bg-purple-500;` on `.editor-post-title `.
Screen Shot 2021-03-26 at 9 54 53 AM

Rebuild example when adding style `@apply bg-yellow-700;` on `body`.
Screen Shot 2021-03-26 at 9 53 49 AM

Initial build load time.
Screen Shot 2021-03-26 at 9 51 52 AM

### OTHER

- [NA] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [NA] Does this pass CBT?

### STEPS TO VERIFY

1. Checkout and build files.
2. Verify that styles and tailwind styles on the frontend work.
3. Verify that no consoles errors are triggered on the frontend.
4. Verify that styles and tailwind styles in the editor work.
5. Verify that no consoles errors are triggered in the editor.

### DOCUMENTATION

Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

Not yet, as this feature is experimental.

### CODE REVIEW

File: inc/scripts.php [hunk 1]
```diff
@@ -5,6 +5,20 @@
...
+/**
+ * Enqueue Tailwind.
+ *
+ * Load Tailwind dependencies in editor and frontend.
+ *
+ * @author WDS
+ */
+function _s_tailwind_scripts() {
+	// Register styles & scripts.
+	wp_enqueue_style( 'tailwind', get_stylesheet_directory_uri() . '/build/tailwind.css', array(), $asset_file['version'] );
```
Comment (coreymcollins) [inline inc/scripts.php:18]: Getting these on the frontend when I load it up:

`Notice: Undefined variable: asset_file in /Applications/MAMP/htdocs/wd_s/wp-content/themes/wd_s/inc/scripts.php on line 17`
`Notice: Trying to access array offset on value of type null in /Applications/MAMP/htdocs/wd_s/wp-content/themes/wd_s/inc/scripts.php on line 17`

**Comment by coreymcollins** [comment]
I know it may vary a bit for everyone, but for me on a 2019 i9 MBP, my compile speeds while running `npm run start` jump from around 2600ms on `main` to around 6600ms on this branch. That's a pretty big jump. I have another example locally that's jumping up to around 5000ms, which still isn't great but is less.

I'll prep a PR for what I'm working on so we can compare/contrast and maybe put some ideas together to land on something a little more efficient than what we've both reached so far.

File: inc/scripts.php (hunk 1 above)
Comment (michealengland) [reply inc/scripts.php:18]: It looks like my `wp debug` wasn't on. I got this resolved.

**Comment by michealengland** [comment]
> I'll prep a PR for what I'm working on so we can compare/contrast and maybe put some ideas together to land on something a little more efficient than what we've both reached so far.

@coreymcollins I think that is a good idea, also are you importing the same things into `editor.scss`? I know that on my original PR I had an empty `editor.scss`. This time around I have the following imports which definitely increase build times.

```scss
//-----------------------------------------
// Editor Files
//-----------------------------------------
@tailwind base;

@import "base/index.scss";
@import "modules/index.scss";
@import "views/index.scss";
```

**Comment by coreymcollins** [comment]
@michealengland you can see what I've got in #617 (just pushed this up) – I'm basically mimicking the theme styles where Tailwind and theme styles are separated into different files, then compiled into a single output file.

**Pull request closed without merging**